msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java

```diff
@@ -586,7 +479,7 @@ public ImmutableValue unpackValue()
         MessageFormat mf = getNextFormat();
         switch (mf.getValueType()) {
             case NIL:
-                unpackNil();
+                readByte();
```

Comment:
Ah, `mf.getValueType() == NIL` means the value is `nil`. Okay.

How about changing https://github.com/msgpack/msgpack-java/pull/310/files#diff-6a54557ce799041aa918fa8fa48bc78dR537 as well?